### PR TITLE
Update modeling_tf_utils.py

### LINKED ISSUE #2906: documentation for TF models mentions non-existent methods
Documentation of `TFPreTrainedModel.from_pretrained` method mentions the `.model()` and `.eval()` methods that are not defined for tensorflow models:

> The model is set in evaluation mode by default using ``model.eval()`` (Dropout modules are deactivated)
> To train the model, you should first set it back in training mode with ``model.train()``

https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_tf_utils.py#L195

### PR DESCRIPTION
**Pull request by BramVanroy**
Tensorflow does not use .eval() vs .train().

closes https://github.com/huggingface/transformers/issues/2906

### CODE REVIEW

**Comment by LysandreJik** [comment]
This is great, thanks @BramVanroy !!
